lib/db_connection/connection_pool.ex

```diff
@@ -40,6 +42,13 @@ defmodule DBConnection.ConnectionPool do
     GenServer.call(pool, {:disconnect_all, interval}, :infinity)
   end
 
+  @doc """
+  Returns connection metrics in the shape of %{active: N, waiting: N}
+  """
+  def get_connection_metrics(pid) do
+    GenServer.call(pid, :get_metrics)
+  end
+
```

Comment:
Now that you're using counters, you don't need to call the connection pool, you can read them directly on this function